packages/experience-builder-sdk/src/hooks/useBreakpoints.ts

```diff
@@ -22,21 +22,23 @@ import { useCallback, useEffect, useState } from 'react';
  * and then decending by screen width. For mobile-first designs, the order would be ascending
  */
 export const useBreakpoints = (breakpoints: Breakpoint[]) => {
-  const [mediaQueryMatchers, initialMediaQueryMatches] = mediaQueryMatcher(breakpoints);
-
+  const fallbackBreakpointIndex = getFallbackBreakpointIndex(breakpoints);
+  const [mediaQueryMatchers, currentQueryMatches] = mediaQueryMatcher(breakpoints);
   const [mediaQueryMatches, setMediaQueryMatches] =
-    useState<Record<string, boolean>>(initialMediaQueryMatches);
+    useState<Record<string, boolean>>(currentQueryMatches);
 
-  const fallbackBreakpointIndex = getFallbackBreakpointIndex(breakpoints);
+  const activeBreakpointIndex = getActiveBreakpointIndex(
+    mediaQueryMatches,
+    fallbackBreakpointIndex
+  );
 
   // Register event listeners to update the media query states
   useEffect(() => {
-    const eventListeners = mediaQueryMatchers.map(({ id, signal }) => {
-      const onChange = () =>
-        setMediaQueryMatches((prev) => ({
-          ...prev,
-          [id]: signal.matches,
-        }));
+    const eventListeners = mediaQueryMatchers.map(({ signal }) => {
```

Comment:
The media query matchers are called on every single stage change of every single breakpoint. So e.g. if I switched from "desktop" to "mobile", you would receive two signal callbacks: one that sets "desktop" to "matches: false" and one that sets "table" to "matches: false". So we would be doing duplicate work by re-calculating everything on every single signal callback.
Can you maybe provide more background on why the previous signals actually did not work?

Reply:
The issue came from state not updating correctly. Basically the `mediaQueryMatches` variable in the original code failed to update and would always have `{ tablet: true }` when selecting the mobile breakpoint. However, `initialMediaQueryMatches` would have the correct latest value. So the key is to always get the latest breakpoint values which happens when you call `mediaQueryMatcher`

Reply:
Yes, you're right. The event listener is called once for mobile and once for tablet. However, after the first call we remove the listener and thus the second one doesn't get triggered. I described this also in the other PR: https://github.com/contentful/experience-builder/pull/339